lib/posix-process/arch/x86_64/execve.c

```diff
@@ -0,0 +1,38 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/arch/ctx.h>
+#include <uk/essentials.h>
+
+void execve_arch_execenv_init(struct ukarch_execenv *execenv_new,
+			      struct ukarch_execenv *execenv,
+			      __uptr ip, __uptr sp)
+{
+	UK_ASSERT(execenv_new);
+	UK_ASSERT(execenv);
+	UK_ASSERT(ip);
+	UK_ASSERT(sp);
+	UK_ASSERT(IS_ALIGNED(sp, UKARCH_SP_ALIGN));
+
+	execenv_new->regs.rip = ip;
+	execenv_new->regs.rsp = sp;
+
+	/* Prepare for iretq
+	 * FIXME re-arch: use GDT macros once moved out of plat/common
+	 */
+	execenv_new->regs.eflags = X86_EFLAGS_IF; /* enable IRQs on ctx switch */
+	execenv_new->regs.cs = 8; /* GDT_DESC_OFFSET(GDT_DESC_CODE) */
+	execenv_new->regs.ss = 16; /* GDT_DESC_OFFSET(GDT_DESC_DATA) */
+
+	/* Copy current ectx to inerhit platform-initialized regs like mxcsr */
+	ukarch_ectx_sanitize((struct ukarch_ectx *)&execenv_new->ectx);
```

Comment:
I assume we do not do this on ARM because there we simply copy the Neon regs which are scratch registers as per ABI anyway? If yes we should for sure leave a comment on the ARM implementation as to why we do not do anything ECTX related there.

Reply:
Yes, we don't copy them though. We leave them in an undefined state similarly to the gpregs.